base/base.jl

```diff
@@ -21,39 +38,79 @@ end
 #    var::Symbol
 #end
 
+"""
+    KeyError(key)
+
+An indexing operation into an `Associative` (`Dict`) or `Set` like object tried to access or
+delete a non-existent element.
+"""
 type KeyError <: Exception
     key
 end
 
+"""
+    MethodError(f, args)
+
+A method with the required type signature does not exist in the given generic function.
+Alternatively, there is no unique most-specific method.
+"""
 type MethodError <: Exception
     f
     args
 end
 
+"""
+    EOFError()
+
+No more data was available to read from a file or stream.
+"""
 type EOFError <: Exception end
 
+"""
+    DimensionMismatch([msg])
+
+The objects called do not have matching dimensionality. Optional argument `msg` is a
+descriptive error string.
+"""
 type DimensionMismatch <: Exception
     msg::AbstractString
 end
 DimensionMismatch() = DimensionMismatch("")
 
+"""
+    AssertionError([msg])
+
+The asserted condition did not evaluate to `true`.
+Optional argument `msg` is a descriptive error string.
+"""
 type AssertionError <: Exception
     msg::AbstractString
-
-    AssertionError() = new("")
     AssertionError(msg) = new(msg)
+    AssertionError() = new("")
```

Comment:
Changing the definition order probably won't affect anything, but best to avoid unrelated changes if possible. Other than that the PR looks go to me, thanks.